### ISSUE TITLE
Different Instance ID with keypoint labeler vs other labelers

### ISSUE THREAD

**Issue by saher-hazen** [Unity-Technologies/com.unity.perception#340]
I have a boundingbox2d labeler, and a keypoint labeler on my perception camera. I have set object filter to be all. I have noticed that 
a - order of annotations is different : I assumed this was expected since a different number of objects may be included in keypoint labeling.
b - instance IDs on the objects are different as well. Problem a led me to match annotations on Instance ID and Label ID. But then I still found mismatches. Digging deeper I was able to identify that instance Ids do not match. 

I am unsure if this is how it is supposed to be or if it's a bug. For now, I am adding prefab name to  a custom version of keypoint labeler so I can match on something other than instance ID. I have multiple objects with the same category hence label ID alone is not sufficient. 

I'll be happy if somebody can point out if I am doing something wrong :)

**Comment by saher-hazen** [comment]
Also, I have noticed cases where external occlusion is less than perfect. I am unsure why this is happening. Be great if someone can verify that it works 100% of the times for them

**Comment by JonathanHUnity** [comment]
The order of the annotations are not necessarily consistent between the labelers, so instance_id is the right way to match up annotations on an object.

We have not seen the instance ID on a single object be different across labelers. That sounds like a bug. Could you provide some  steps to reproduce this issue? Or, could you send us your project for us to investigate? Feel free to reach out to computer-vision@unity3d.com for a secure link to upload your project.

Thanks.

**Comment by saher-hazen** [comment]
Hi @JonathanHUnity, thanks for getting back to me. I have remained unable to reproduce this issue in a clean set up. In variants of my previous project, it does appear consistently, but I can't find a reason. So I retraced my steps and set a project up from scratch, importing assets incrementally. I have not seen it pop up since. If it does and if I can reliably reproduce it, I'll be sure to add more detail here.
